File: cs/Refactor1/SRP_DI_Workshop/ServiceInterface/ResponseMessageFactory.cs
using System;
using System.Net;
using SRP_DI_Workshop.Domain;

namespace SRP_DI_Workshop.ServiceInterface
{
    public class ResponseMessageFactory
    {
        public ResponseMessage CreateCancellationResponseMessage(RequestMessage reqMsg, Order orderToCancel, string message)
        {
            return new ResponseMessage
            {
                StatusCode = HttpStatusCode.OK,
                ResponseBody = new OrderCancellationResponse
                {
                    RequestId = reqMsg.RequestId,
                    OrderId = reqMsg.OrderId,
                    OrderState = orderToCancel.State.ToString(),
                    Message = message
                }
            };
        }

        public ResponseMessage CreateOrderSubmissionResponseMessage(RequestMessage reqMsg, Order order)
        {
            return new ResponseMessage
            {
                StatusCode = HttpStatusCode.OK,
                ResponseBody = new OrderSubmissionResponse
                {
                    RequestId = reqMsg.RequestId,
                    OrderId = order.OrderId,
                    OrderState = order.State.ToString()
                }
            };
        }

        public ResponseMessage CreateOrderQueryResponseMessage(RequestMessage reqMsg, Order orderToGet)
        {
            return new ResponseMessage
            {
                StatusCode = HttpStatusCode.OK,
                ResponseBody = new OrderQueryResponseMessage
                {
                    RequestId = reqMsg.RequestId,
                    Order = orderToGet
                }
            };
        }

        public ResponseMessage CreateErrorResponseMessage(Exception ex)
        {
            return new ResponseMessage
            {
                StatusCode = HttpStatusCode.InternalServerError,
                ResponseBody = ex.ToString()
            };
        }

        public ResponseMessage CreateBadRequestResponseMessage(RequestMessage reqMsg)
        {
            return new ResponseMessage 
            {
                StatusCode = HttpStatusCode.BadRequest,
                ResponseBody = reqMsg 
            };
        }
    }
}

File: java/Refactor2/src/main/java/no/bouvet/solid/srpdip/messageinterface/ResponseMessageFactory.java
package no.bouvet.solid.srpdip.messageinterface;

import no.bouvet.solid.srpdip.Factory;
import no.bouvet.solid.srpdip.domain.Order;

public class ResponseMessageFactory {

	public static Factory<ResponseMessageFactory> factory = new Factory<>(ResponseMessageFactory.class);

	public ResponseMessage createOrderQueryResponseMessage(RequestMessage reqMsg, Order orderToGet)
	{
		return new OrderQueryResponseMessage(reqMsg.getRequestId(), orderToGet);
	}

	public ResponseMessage createCancellationResponseMessage(RequestMessage reqMsg, Order orderToCancel, String message)
	{
		return new OrderCancellationResponse(
				reqMsg.getRequestId(),
				reqMsg.getOrderId(),
				orderToCancel.getState().toString(),
				message);
	}

	public ResponseMessage createErrorResponseMessage(long requestId, Exception ex)
	{
		return new ResponseMessage(requestId);
	}

	public ResponseMessage createOrderSubmissionResponseMessage(RequestMessage reqMsg, Order order)
	{
		return new OrderSubmissionResponse(
				reqMsg.getRequestId(),
				order.getOrderId(),
				order.getState().toString());
	}
}
